### ISSUE TITLE
Failed to compile on macOS 10.15.3

### ISSUE THREAD

**Issue by f32by** [EQMG/Acid#92]
**Describe the bug**
```
1. compiler complains about missing including <unordered_map>.

2. ld: can't map file, errno=22 file '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/OpenAL.framework' for architecture x86_64
```

**To Reproduce**
1. Follow the build instructions in [this wiki page](https://github.com/EQMG/Acid/wiki/Setup-MacOS)
2. Open Acid.xcodeproj
3. Build target TestTourial1

**Expected behaviour**
build should succeed

**Screenshots**
No

**Hardware:**
 - Device: MacBook Pro 16-inch
 - OS: macOS 10.15.3

**Additional context**
I noticed [this commit](https://github.com/EQMG/Acid/commit/a7e29acee4229e52fd07e3411dab017f81232e67) fixes this issue but build still fails :(

**Comment by mattparks** [comment]
Looks like I just forgot to include <unordered_map> in a few spots, and msdf-c inline methods also failed to be found. Also in macOS 10.14+ releases OpenAL soft must be installed. I made those changes here: https://github.com/EQMG/Acid/commit/29320ed3d74af9c7be2703bc3e2362a4d923b157

Fixed to macOS are not complete yet, the upgrade from GLFW 3.3.1 to 3.3.2 has broke `glfwCreateWindowSurface`, and DPI scaling on macOS displays the engine at the wrong resolution.
